pkg/kubelet/kubeletconfig/util/files/files_test.go

```diff
@@ -0,0 +1,339 @@
+		},
+		"exists as dir": {
+			path:         prePath + "/ExistsAsDir",
+			isFile:       false,
+			create:       true,
+			expectedBool: true,
+			expectedErr:  nil,
+		},
+	}
+
+	for testcaseName, testcase := range testcases {
+		t.Run(testcaseName, func(t *testing.T) {
+			if testcase.create {
+				if err := createFileOrDir(testcase.isFile, fs, testcase.path); err != nil {
+					t.Fatalf("unexpected error in creating %s, error: %v", testcase.path, err)
+				}
+			}
+
+			if result, err := DirExists(fs, testcase.path); err != nil {
+				// if the path is file, should get error
+				if !reflect.DeepEqual(testcase.expectedErr, err) {
+					t.Errorf("unexpected error in creating dir, expected: %v, actual: %v", testcase.expectedErr, err)
+				}
+			} else {
+				if testcase.expectedBool != result {
+					t.Errorf("unexpected result, expected: %v, actual: %v", testcase.expectedBool, result)
+				}
+			}
+		})
+
+	}
+}
+
+func TestEnsureDir(t *testing.T) {
+	prePath := getTempDir(t, "TestEnsureDir")
+	fs := utilfs.DefaultFs{}
+	defer cleanup(t, fs, prePath)
+
+	testcases := map[string]struct {
+		path        string // path to be created
+		isFile      bool   // if the path indicates a file
+		create      bool   // if should create dir before test
+		expectedErr error
+	}{
+		"exists as file": {
+			path:        prePath + "/ExistsAsFile",
+			isFile:      true,
+			create:      true,
+			expectedErr: fmt.Errorf("expected dir at %q, but mode is %q", prePath + "/ExistsAsFile", "-rw-r--r--"), // should have error, but the error info can only be fulfilled after creating the file
+		},
+		"exists as dir": {
+			path:        prePath + "/NotExists",
+			isFile:      false,
+			create:      true,
+			expectedErr: nil,
+		},
+		"not exists in sub dir": {
+			path:        prePath + "/SubDir1/SubDir2/Exists",
+			isFile:      false,
+			create:      false,
+			expectedErr: nil,
+		},
+	}
+
+	for testcaseName, testcase := range testcases {
+		t.Run(testcaseName, func(t *testing.T) {
+			if testcase.create {
+				if err := createFileOrDir(testcase.isFile, fs, testcase.path); err != nil {
+					t.Fatalf("unexpected error in creating %s, error: %v", testcase.path, err)
+				}
+			}
+
+			// ensure dir exists
+			if err := EnsureDir(fs, testcase.path); err != nil {
+				// if the path is file, should get error
+				if !reflect.DeepEqual(testcase.expectedErr, err) {
+					t.Errorf("unexpected error in creating dir, expected: %v, actual: %v", testcase.expectedErr, err)
+				}
+			} else {
+				// check if dir created
+				info, err := fs.Stat(testcase.path)
+				if err != nil {
+					t.Errorf("unexpected error in verifying dir: %s, error: %v", testcase.path, err)
+				}
+				if !info.IsDir() {
+					t.Errorf("expected dir in path: %q, but got: %v", testcase.path, info.Mode())
+				}
+			}
+		})
+	}
+}
+
+func createFileOrDir(isFile bool, fs utilfs.Filesystem, path string) error {
+	if isFile {
+		return createFile(fs, path)
+	} else {
```

Comment:
golint complaining about this

pkg/kubelet/kubeletconfig/util/files/files_test.go:301:9: if block ends with a return statement, so drop this else and outdent its block